src/redux/subscriptions/reducer.js

```diff
@@ -1,19 +1,78 @@
 import * as actionConstant from './constants';
 
 const INITIAL_STATE = {
-  data: [],
-  error: '',
+  subs: [],
+  sub: { classId: '', memberId: '', date: '' },
+  error: false,
+  message: '',
+  id: '',
   isPending: false
 };
 
 const reducer = (state = INITIAL_STATE, action) => {
   switch (action.type) {
-    case actionConstant.GET_SUBSCRIPTIONS_PENDING:
-      return { ...state, isPending: !state.isPending };
+    case actionConstant.RESET_STATE:
+      return {
+        ...state,
+        sub: { classId: '', memberId: '', date: '' },
+        error: false,
+        id: '',
+        isPending: false
+      };
+    case actionConstant.SUBSCRIPTIONS_PENDING:
+      return { ...state, isPending: true };
+    case actionConstant.SELECT_ID:
+      return { ...state, id: action.payload };
     case actionConstant.GET_SUBSCRIPTIONS_SUCCESS:
-      return { ...state, data: action.payload };
+      return { ...state, subs: action.payload, isPending: false, error: false };
     case actionConstant.GET_SUBSCRIPTIONS_ERROR:
-      return { ...state, error: action.payload };
+      return { ...state, message: action.payload, isPending: false, error: true };
+    case actionConstant.DELETE_SUBSCRIPTIONS_SUCCESS:
+      return {
+        ...state,
+        id: '',
+        message: action.payload.message,
+        isPending: false,
+        subs: state.subs.filter((subs) => subs._id !== action.payload.id),
+        error: false
+      };
+    case actionConstant.DELETE_SUBSCRIPTIONS_ERROR:
+      return {
+        ...state,
+        id: '',
+        isPending: false,
+        message: action.payload,
+        error: true
+      };
+    case actionConstant.POST_SUBSCRIPTIONS_SUCCESS:
+      return {
+        ...state,
+        isPending: false,
+        message: action.payload.message,
+        subs: [...state.subs, action.payload.newSubscription]
+      };
+    case actionConstant.POST_SUBSCRIPTIONS_ERROR:
+      return {
+        ...state,
+        isPending: false,
+        message: action.payload,
+        error: true
+      };
+    case actionConstant.PUT_SUBSCRIPTIONS_SUCCESS:
+      state.subs.filter((sub) => sub._id !== action.payload.id);
```

Comment:
Put endpoint it's not working correctly, you should update the item in it's index to update the array global state, if you test it, it works one yes and one no.

Reply:
I don't experience that issue, there I simply change 'subs' and the 'useEffect' fetches the updated array from the database. So, it wouldn't be necessary to update it right there. However, I understand that it would be more correct to add it in that part of the code. So... fixed 👍